server/aesdsocket.c

```diff
@@ -0,0 +1,357 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+
+#define PORT_NUM 9000
+#define BUFSIZE 30000
+
+int stop_flag;
+
+void signal_handler(int signal_num);
+int sendall(int s, char *buf, int *len);
+
+int main(int argc, char * argv[]){
+
+	int sockfd, n, clientfd;
+	struct sockaddr_in serveraddr; /* server's addr */
+  	struct sockaddr_in clientaddr; /* client addr */
+  	struct hostent *hostp; /* client host info */
+	//static char buf[BUFSIZE];
+	char *hostaddrp;
+	struct sigaction sa;
+	stop_flag = 0;
+
+	char * buf = malloc(BUFSIZE);
+
+	if(buf == NULL){
+		perror("error with malloc");
+		return -1;
+	}
+
+	fflush(stdin);
+
+	sa.sa_handler = signal_handler;
+
+	if (sigaction(SIGINT, &sa, NULL) == -1) {
+        perror("Error setting up SIGINT handler");
+        return -1;
+    }
+    if (sigaction(SIGTERM, &sa, NULL) == -1) {
+        perror("Error setting up SIGTERM handler");
+        return -1;
+    }
+
+	openlog("aesdsocket", LOG_PID, LOG_USER);
+
+	if(argc == 1){
+		sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+  		if (sockfd < 0){
+			perror("ERROR opening socket");
+			closelog();
+			return -1;
+		}
+
+		bzero((char *) &serveraddr, sizeof(serveraddr));
+		serveraddr.sin_family = AF_INET;
+		serveraddr.sin_addr.s_addr = htonl(INADDR_ANY);
+		serveraddr.sin_port = htons((unsigned short)PORT_NUM);
+
+		if (bind(sockfd, (struct sockaddr *) &serveraddr, sizeof(serveraddr)) < 0){
+    		perror("ERROR on binding");
+			closelog();
+			return -1;
+		}
+
+		int clientlen = sizeof(clientaddr);
+
+		listen(sockfd , 3);
+		
+		int c = sizeof(struct sockaddr_in);
+
+		while(1 && !stop_flag){
+			clientfd = accept(sockfd, (struct sockaddr *)&clientaddr, (socklen_t*)&c);
+
+			hostaddrp = inet_ntoa(clientaddr.sin_addr);
+
+			syslog(LOG_INFO, "Accepted connection from %s", hostaddrp);
+
+			int fd = open("/var/tmp/aesdsocketdata", O_CREAT|O_RDWR, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+
+			if(fd == -1){
+				perror("error with open");
+				closelog();
+				return -1;
+			}
+
+			while(1){
+				bzero(buf, BUFSIZE);
+				n = recv(clientfd, buf, BUFSIZE, 0);
+				if (n < 0){
+					perror("ERROR in recv");
+					closelog();
+					close(sockfd);
+					return -1;
+				}
+				printf("read: %d bytes\n", n);
+				char *token;
+				int found_terminator = 0;
+
+				for(int i = 0; i < n; i++){
+					if(buf[i] == '\n'){
+						found_terminator = 1;
+					}
+				}
+
+				if((token = strtok(buf, "\n")) == NULL){
```

Comment:
What's the reason for having two different ways to find `\n`?  Consider using one method for this.